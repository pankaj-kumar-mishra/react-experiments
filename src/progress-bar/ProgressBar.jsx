import React, { useState, useEffect } from "react";
import styles from "./ProgressBar.module.css";

const ProgressBar = ({ percent, text = "Please wait..." }) => {
  const [width, setWidth] = useState(50);

  useEffect(() => {
    setWidth(percent);
  }, [percent]);

  return (
    <div className={styles.progressBar}>
      <div
        role="progressbar"
        aria-valuemin="0"
        aria-valuemax="100"
        aria-valuenow={width}
        className={styles.line}
        style={{
          width: `${width}%`,
        }}
      ></div>
      {!!width && (
        <h4 data-testid="progressBarMsg" className={styles.text}>
          {text}
        </h4>
      )}
    </div>
  );
};

export default ProgressBar;
