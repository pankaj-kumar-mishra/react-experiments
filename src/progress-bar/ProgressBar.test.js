import React from "react";
import { render, screen } from "@testing-library/react";
import ProgressBar from "./ProgressBar";

describe("ProgressBar", () => {
  it("should match snapshot", () => {
    const { asFragment } = render(
      <ProgressBar percent={40} text="Fetching..." />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("should renders progress bar with correct width", () => {
    render(<ProgressBar percent={60} text="60% Complete" />);
    const progressBar = screen.getByRole("progressbar");
    expect(progressBar.style.width).toBe("60%");
  });

  it("should render progress bar message if width exists", () => {
    render(<ProgressBar percent={40} text="40% Complete" />);
    const progressBarMsg = screen.getByTestId("progressBarMsg");
    expect(progressBarMsg.textContent).toBe("40% Complete");
  });

  it("should not render progress bar message if width is falsy", () => {
    render(<ProgressBar percent={0} text="40% Complete" />);
    const progressBarMsg = screen.queryByTestId("progressBarMsg");
    expect(progressBarMsg).toBeNull();
  });
  it("should not render progress bar message if text is empty", () => {
    render(<ProgressBar percent={40} text="" />);
    const progressBarMsg = screen.queryByTestId("progressBarMsg");
    expect(progressBarMsg).toBeNull();
  });
});
