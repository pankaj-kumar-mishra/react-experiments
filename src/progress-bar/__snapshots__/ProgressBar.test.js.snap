// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="progressBarContainer"
  >
    <div
      class="progressBar"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="40"
        class="line"
        role="progressbar"
        style="width: 40%;"
      />
      <h4
        class="text"
        data-testid="progressBarMsg"
      >
        Fetching...
      </h4>
    </div>
  </div>
</DocumentFragment>
`;
